src/dawn/SIR/SIR.proto

```diff
@@ -0,0 +1,627 @@
+/*===------------------------------------------------------------------------------*- proto -*-===*\
+ *                          _
+ *                         | |
+ *                       __| | __ ___      ___ ___
+ *                      / _` |/ _` \ \ /\ / / '_  |
+ *                     | (_| | (_| |\ V  V /| | | |
+ *                      \__,_|\__,_| \_/\_/ |_| |_| - Compiler Toolchain
+ *
+ *
+ *  This file is distributed under the MIT License (MIT).
+ *  See LICENSE.txt for details.
+ *
+\*===------------------------------------------------------------------------------------------===*/
+
+// @defgroup sir_proto SIR protobuf
+// @brief This is a SIR description for Google's protocol buffers library
+// <a href="https://developers.google.com/protocol-buffers/">Protobuf</a>.
+
+syntax = "proto3";
+
+package dawn.sir.proto;
+
+// @brief Source information
+//
+// `(-1,-1)` indicates an invalid location.
+// @ingroup sir_proto
+message SourceLocation {
```

Comment:
do we need also a "file"?

Reply:
At moment we are under the assumption that everything is in the same file (thus only the line and col is required). If we ever drop this requirement we may have to store a file index in the `SourceLocation` and keep a map of these file indices to the actual filenames in the SIR.